pkg/operator/controller/controller.go

```diff
@@ -461,17 +458,19 @@ func (r *reconciler) ensureDNS(dns *operatorv1.DNS) error {
 // in the source ca bundle configmap (e.g. cert rotation). Apart from resource versions,
 // server name is also used in the path of ca bundles in case the same ca bundle configmap
 // is specified for two different servers.
-func (r *reconciler) caBundleRevisionMap(dns *operatorv1.DNS) (map[string]string, error) {
+func (r *reconciler) caBundleRevisionMap(dns *operatorv1.DNS) map[string]string {
 	caBundleRevisions := map[string]string{}
 	transportConfig := dns.Spec.UpstreamResolvers.TransportConfig
 	if transportConfig.Transport == operatorv1.TLSTransport {
 		if transportConfig.TLS != nil && transportConfig.TLS.CABundle.Name != "" {
 			name := CABundleConfigMapName(transportConfig.TLS.CABundle.Name)
 			cm := &corev1.ConfigMap{}
-			if err := r.client.Get(context.TODO(), name, cm); err != nil {
-				return caBundleRevisions, err
+			err := r.client.Get(context.TODO(), name, cm)
+			if err != nil {
+				logrus.Warningf("failed to get destination ca bundle configmap %s: %v", name.Name, err)
```

Comment:
Can you explain the difference between returning an `err` and this `logrus`? I understand the standard `err`, that just gets thrown in the cluster-dns-operator logs, but where does `logrus` go?

Reply:
We don't return an error in this case because we don't want to stop execution if a configmap doesn't exist. `logrus` will send a record to the logs using whatever the operator has configured; as far as I can tell, the operator doesn't have any special logging configuration so it should be going to stdout.

Reply:
Ah got it. So returning an error actually can change code flow, but you just want to throw an warning and continue on if there is no valid configmap.

One thought: If I create an upstream configuration to use a configmap `foo`, but `foo` isn't created yet, but then I create `foo` after I create the configmap, what will happen? Will it ignore the newly create configmap `foo`? Or will it reconcile at some point and use `foo` accordingly?

Reply:
Good question! The reconcile loop will keep looking for the source CM and logging the warning until it's created. Once the source CM is created, it'll update the daemonset and corresponding Corefile. While the source configmap is missing, you'll see these in the operator logs:
```
time="2022-06-07T17:10:35Z" level=warning msg="source ca bundle configmap missing-cm does not exist"
time="2022-06-07T17:10:35Z" level=warning msg="failed to get destination ca bundle configmap ca-missing-cm: configmaps \"ca-missing-cm\" not found"
```
Once the source configmap is created you'll see it resolve right away:
```
time="2022-06-07T17:16:06Z" level=info msg="created configmap openshift-dns/ca-missing-cm"
time="2022-06-07T17:16:06Z" level=info msg="updated dns daemonset openshift-dns/dns-default:   &v1.DaemonSet
```
`ca-missing-cm` is the destination CM that the operator creates by making a copy of the source CM and then the next log is the daemonset/Corefile update.

Reply:
Got it. Thanks for the detailed answer!